### ISSUE TITLE
Feature request: --fail-level info

### ISSUE THREAD

**Issue by hpclj** [clj-kondo/clj-kondo#2609]
**Is your feature request related to a problem? Please describe.**

We use clj-kondo as part of our pipeline to prevent us from merging code that's not linted. We like clj-kondo so much that we want to apply not only the warnings and errors, but also useful info messages. eg:

for `(str (str "a") "b")` it gives something like `redundant str` as an `info` message

**Describe the solution you'd like**
To exit with non-zero status when the output isn't simply just
`linting took 567ms, errors: 0, warnings: 0`

**Describe alternatives you've considered**
To use bash-foo and diy parse the output to determinte if there were info messages

**Comment by borkdude** [comment]
This gives two warnings:

```
$ clj-kondo --lint - <<< '(str (str "a") "b")'
<stdin>:1:6: info: Redundant nested call: str
<stdin>:1:6: warning: Single argument to str already is a string
```

One at level warning and one at level info. So the exit code should have been non-zero already?

For `:info` level messages: clj-kondo chooses this level for linters that might be informative but could also be a false positive.
You can choose the level of each linter yourself using configuration. If you think nested str should be a warning, why not tweak the level yourself?

**Comment by hpclj** [comment]
Thanks for the quick response!

Odd, the same example doesn't work on my machine `v2025.07.28`

```
➜ clj-kondo --lint - <<< '(str (str "a") "b")'
<stdin>:1:6: info: Redundant nested call: str
<stdin>:1:6: info: Single argument to str already is a string
linting took 7ms, errors: 0, warnings: 0
➜ echo $?
0
```

What do you mean by I can tweak the level myself, are you referring to the `config.edn` in `.clj-kondo` ?

**Comment by borkdude** [comment]
Ah, this is because I have set this in my home configuration:

```
:redundant-str-call {:level :warning}
```

> What do you mean by I can tweak the level myself, are you referring to the config.edn in .clj-kondo ?

Yep!

**Comment by hpclj** [comment]
thanks for the info! 

I now understand why this can't be done -> because of potential false positives.

**Comment by borkdude** [comment]
Not all the the info level linters result in false positives, but some may, like `redundant-ignore`

**Comment by hpclj** [comment]
Good to know thank you. As a new clojure user unfamiliar with clj-kondo, I can't differentiate which are "good" info messages or false positives, so I'll just ignore them all for the time being.

**Comment by borkdude** [comment]
You could also consider `:info` level messages as non-urgent messages that don't affect your code too much, that you'd just want to correct while you're touching the code anyway.

**Comment by hpclj** [comment]
Yea, what I meant was that I'd still get my feedback to catch and/or learn things with clj-kondo, but on the pipeline I'd ignore them :)
